charts/truefoundry/templates/tfy-nginx-proxy/_helpers.tpl

```diff
@@ -0,0 +1,276 @@
+{{- default "tfy-nginx-proxy" .Values.tfyNginxProxy.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+  Create a default fully qualified app name.
+  We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+  If release name contains chart name it will be used as a full name.
+  */}}
+{{- define "tfy-nginx-proxy.fullname" -}}
+{{- if .Values.tfyNginxProxy.fullnameOverride }}
+{{- .Values.tfyNginxProxy.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default "tfy-nginx-proxy" .Values.tfyNginxProxy.nameOverride }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+  Create chart name and version as used by the chart label.
+  */}}
+{{- define "tfy-nginx-proxy.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+  Pod Labels
+  */}}
+{{- define "tfy-nginx-proxy.podLabels" -}}
+{{ include "tfy-nginx-proxy.selectorLabels" . }}
+{{- if .Values.tfyNginxProxy.image.tag }}
+app.kubernetes.io/version: {{ .Values.tfyNginxProxy.image.tag | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- range $name, $value := .Values.tfyNginxProxy.commonLabels }}
+{{ $name }}: {{ tpl $value $ | quote }}
+{{- end }}
+{{- end }}
+
+{{/*
+  Common labels
+  */}}
+{{- define "tfy-nginx-proxy.labels" -}}
+helm.sh/chart: {{ include "tfy-nginx-proxy.chart" . }}
+{{ include "tfy-nginx-proxy.podLabels" . }}
+{{- if .Values.tfyNginxProxy.commonLabels }}
+{{ toYaml .Values.tfyNginxProxy.commonLabels }}
+{{- else if .Values.global.labels }}
+{{ toYaml .Values.global.labels }}
+{{- end }}
+{{- end }}
+
+{{/*
+  Common annotations
+  */}}
+{{- define "tfy-nginx-proxy.annotations" -}}
+{{- if .Values.tfyNginxProxy.annotations }}
+{{ toYaml .Values.tfyNginxProxy.annotations }}
+{{- else if .Values.global.annotations }}
+{{ toYaml .Values.global.annotations }}
+{{- else }}
+{}
+{{- end }}
+{{- end }}
+
+{{/*
+  Deployment annotations
+  */}}
+{{- define "tfy-nginx-proxy.deploymentAnnotations" -}}
+{{- $merged := merge (dict "argocd.argoproj.io/sync-wave" "3") (include "tfy-nginx-proxy.annotations" . | fromYaml) }}
+{{- toYaml $merged }}
+{{- end }}
+
+{{/*
+  Pod annotations
+*/}}
+{{- define "tfy-nginx-proxy.podAnnotations" -}}
+{{- /* User-provided pod annotations from values */ -}}
+{{- $podVals := .Values.tfyNginxProxy.podAnnotations | default (dict) -}}
+
+{{- /* Render the ConfigMap template and hash it */ -}}
+{{- $cm := include (print $.Template.BasePath "/tfy-nginx-proxy/configmap.yaml") . -}}
+
+{{- $_ := set $podVals "checksum/config" (sha256sum $cm) -}}
+
+{{- toYaml $podVals -}}
+{{- end -}}
+
+{{/*
+  Service Account Annotations
+  */}}
+{{- define "tfy-nginx-proxy.serviceAccountAnnotations" -}}
```

Comment:
Let's add a serviceaccount file. by default the serviceaccount will be create=false and defaults to `default` serviceaccount in the namespace.